service/src/test/java/bio/terra/pipelines/controller/JobsApiControllerTest.java

```diff
@@ -0,0 +1,197 @@
+import bio.terra.pipelines.service.JobsService;
+import bio.terra.pipelines.service.PipelinesService;
+import bio.terra.pipelines.service.model.Job;
+import bio.terra.pipelines.testutils.MockMvcUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.time.Instant;
+import java.util.*;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+@ContextConfiguration(classes = {JobsApiController.class, GlobalExceptionHandler.class})
+@WebMvcTest()
+class JobsApiControllerTest {
+  @MockBean JobsService jobsServiceMock;
+  @MockBean PipelinesService pipelinesServiceMock;
+  @MockBean SamUserFactory samUserFactoryMock;
+  @MockBean BearerTokenFactory bearerTokenFactory;
+  @MockBean SamConfiguration samConfiguration;
+  @MockBean SamService samService;
+
+  @Autowired private MockMvc mockMvc;
+  private final SamUser testUser =
+      new SamUser(
+          "test@email",
+          UUID.randomUUID().toString(),
+          new BearerToken(UUID.randomUUID().toString()));
+
+  private final String pipelineId = "imputation";
+  private final Instant timestamp = Instant.now();
+  private final UUID jobIdOkDone = UUID.randomUUID();
+  private final UUID secondJobId = UUID.randomUUID();
+  private final Job jobOkDone =
+      new Job(
+          jobIdOkDone,
+          testUser.getSubjectId(),
+          pipelineId,
+          "v0",
+          timestamp,
+          Optional.of(timestamp),
+          "COMPLETED");
+  private final Job secondJob =
+      new Job(
+          secondJobId,
+          testUser.getSubjectId(),
+          pipelineId,
+          "v0",
+          timestamp,
+          Optional.of(timestamp),
+          "COMPLETED");
+
+  @BeforeEach
+  void beforeEach() {
+    when(samUserFactoryMock.from(any(HttpServletRequest.class), any())).thenReturn(testUser);
+  }
+
+  @Test
+  void testGetMessageOk() throws Exception {
+
+    when(jobsServiceMock.getJob(testUser.getSubjectId(), pipelineId, jobIdOkDone.toString()))
+        .thenReturn(jobOkDone);
+
+    MvcResult result =
+        mockMvc
+            .perform(get(String.format("/api/jobs/v1alpha1/%s/%s", pipelineId, jobIdOkDone)))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+            .andReturn();
+
+    ApiGetJobResponse response =
+        new ObjectMapper()
+            .readValue(result.getResponse().getContentAsString(), ApiGetJobResponse.class);
+    // you could compare other fields here too beyond the id, if wanted
```

Comment:
I would think we would want to assert that all the expected fields are present in the response. any reason not to do this?